### PR TITLE
Feat/new search

### PR DESCRIPTION
**Pull request by jennieramida**

### CODE REVIEW

**Comment by tansawit** [comment]
small bug. most likely API level?

CleanShot 2567-07-04 at 12 28 20@2x

File: src/lib/layout/SearchResultRenderer.tsx [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+  metadata,
+  value,
+  type,
+  isAccountAddress,
+}: {
+  metadata: ResultMetadata;
+  value: string;
+  type: SearchResultType;
+  isAccountAddress: boolean;
+}) => {
```
Comment (Poafs1) [inline src/lib/layout/SearchResultRenderer.tsx:98]: Can you move this to interface pls.

File: src/lib/layout/SearchResultRenderer.tsx [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+  onClose,
+}: {
+  results: SearchResultType[];
+  keyword: string;
+  cursor: Option<number>;
+  metadata: ResultMetadata;
+  setCursor: (index: Option<number>) => void;
+  handleSelectResult: (type?: SearchResultType, isClick?: boolean) => void;
+  onClose?: () => void;
+}) => (
```
Comment (Poafs1) [inline src/lib/layout/SearchResultRenderer.tsx:278]: ditto

**Review by Poafs1 — APPROVED** [review]
lgtm. pls wait for Nat and Peach to review before merge

File: src/lib/layout/search/SearchComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+  );
+
+  useOutsideClick({
+    ref: boxRef,
+    handler: onClose,
+  });
+
+  useEaster(keyword);
+
+  return (
```
Comment (evilpeach) [inline src/lib/layout/search/SearchComponent.tsx:191]: Change to this

```ts
if (isMobile) return (<Button variant="outline-gray" size="sm" onClick={onOpen}>
          <CustomIcon name="search" boxSize={3} />
        </Button>)

return (<Flex>...</Flex>)
```

File: src/lib/layout/search/SearchZeroState.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+export const SearchZeroState = ({
+  isWasm,
+  isPool,
+  isMove,
+  isGov,
+  isFullTier,
+}: SearchZeroStateProps) => {
+  const starter = ["Account Address", "TX Hash", "Block Height"];
```
Comment (evilpeach) [inline src/lib/layout/search/SearchZeroState.tsx:18]: starter => base

File: src/lib/layout/search/SearchZeroState.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      <Flex
+        direction="column"
+        px={4}
+        py={2}
+        border="1px solid"
+        borderColor="gray.700"
+        borderRadius={8}
+      >
+        {supportedItemsType.map((item) => (
+          <Flex alignItems="center" gap={3}>
```
Comment (evilpeach) [inline src/lib/layout/search/SearchZeroState.tsx:43]: Please add key.

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: src/lib/layout/search/SearchComponent.tsx (hunk 3 above)
Comment (jennieramida) [reply src/lib/layout/search/SearchComponent.tsx:191]: can't, use same modal below

File: src/lib/layout/search/SearchDisplayResult.tsx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+} from "lib/services/searchService";
+
+interface DisplayResultProps {
+  metadata: ResultMetadata;
+  value: string;
+  type: SearchResultType;
+  isAccountAddress: boolean;
+}
+
+export const SearchDisplayResult = ({
```
Comment (songwongtp) [inline src/lib/layout/search/SearchDisplayResult.tsx:16]: SearchResultItemBody

**Review by songwongtp — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Review by Poafs1 — APPROVED** [review]

**Pull request merged**